### PR TITLE
[HSEARCH] 7.1.0.CR1 Release

### PR DESCRIPTION
**Pull request by marko-bekhta**
https://staging.in.relation.to/2024/02/12/hibernate-search-7-1-0-CR1/

### CODE REVIEW

File: posts/MarkoB/2023-11-30-hibernate-search-7-1-0-Alpha1.adoc [hunk 1]
```diff
@@ -3,6 +3,8 @@ Marko Bekhta
 :awestruct-tags: [ "Hibernate Search", "Lucene", "Elasticsearch", "Releases" ]
 :awestruct-layout: blog-post
 :hsearch-doc-url-prefix: https://docs.jboss.org/hibernate/search/7.1/reference/en-US/html_single/
+:hsearch-getting-started-orm-url-prefix: https://docs.jboss.org/hibernate/search/7.1/getting-started/orm/en-US/html_single/
+:hsearch-getting-started-stanadlone-url-prefix: https://docs.jboss.org/hibernate/search/7.1/getting-started/standalone/en-US/html_single/
```
Comment (yrodiere) [inline posts/MarkoB/2023-11-30-hibernate-search-7-1-0-Alpha1.adoc]: Typo: stanadlone => stanadlone.

Same throughout the whole PR :)

File: posts/MarkoB/2024-02-12-hibernate-search-7-1-0-CR1.adoc [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+:hsearch-jira-project-id: 10061
+:hsearch-jira-version-id: 32251
+---
+
+We just published Hibernate Search 7.1.0.CR1,
+a first candidate release of the next minor version of Hibernate Search.
+
+This version brings a lot of updates and improvements for working with a Standalone POJO mapper,
+new query string predicate,
+as well as takes the advantage of the new Elasticsearch version and removes some of the limitations on vector search capabilities.
```
Comment (yrodiere) [inline posts/MarkoB/2024-02-12-hibernate-search-7-1-0-CR1.adoc]:
```suggestion
and takes advantage of Elasticsearch 8.12 to remove some of the limitations on vector search capabilities.
```

File: posts/MarkoB/2024-02-12-hibernate-search-7-1-0-CR1.adoc [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+=== Dependency upgrades
+
+[[orm-version]]
+Hibernate ORM (link:{hsearch-jira-url-prefix}/HSEARCH-5078[HSEARCH-5078])::
+Hibernate Search now depends on Hibernate ORM 6.4.4.Final.
+[[lucene-version]]
+Lucene (link:{hsearch-jira-url-prefix}/HSEARCH-5069[HSEARCH-5069])::
+The Lucene backend now uses Lucene 9.9.2.
+Which brings some bug fixes related to vector search.
```
Comment (yrodiere) [inline posts/MarkoB/2024-02-12-hibernate-search-7-1-0-CR1.adoc]:
```suggestion
The Lucene backend now uses Lucene 9.9.2,
which brings some bug fixes related to vector search.
```

File: posts/MarkoB/2024-02-12-hibernate-search-7-1-0-CR1.adoc [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+SearchScope<Person> personSubTypesScope = searchMapping.scope( Person.class,
+        List.of( "Manager", "Associate" ) ); // <3>
+----
+<1> link:{hsearch-doc-url-prefix}#entrypoints-search-mapping[Retrieve the `SearchMapping`].
+<2> Create a `SearchScope` targeting only the `Book` entity type using a class.
+<3> Create a `SearchScope` targeting only the `Manager` and `Associate` subtypes using their entity names.
+====
+
+See this section of the reference documentation on the link:{hsearch-doc-url-prefix}#entrypoints-search-scope[ways to create search scope]
+for more information.
```
Comment (yrodiere) [inline posts/MarkoB/2024-02-12-hibernate-search-7-1-0-CR1.adoc]: Not sure this is worth mentioning... maybe move it last?

File: posts/MarkoB/2024-02-12-hibernate-search-7-1-0-CR1.adoc [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+** another boolean query constructed from `(crime || investigation || disappearance)` string with a boost of `10`
+** a query matching the phrase `investigation help` with the phrase slop equals to `2`
+* a must not clause matching a regular expression `(dis)?a[p]+ea?ance`
+
++
+Note that each of the mentioned clauses may itself end up being translated into other types of queries.
+====
+
+See this section of the reference documentation on the link:{hsearch-doc-url-prefix}#search-dsl-predicate-query-string[`queryString` predicate]
+for more information.
```
Comment (yrodiere) [inline posts/MarkoB/2024-02-12-hibernate-search-7-1-0-CR1.adoc]: These are the most important points IMO, I'd move them first.

File: posts/MarkoB/2024-02-12-hibernate-search-7-1-0-CR1.adoc [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+
+[[search-dsl-predicate-knn]]
+=== `knn` predicate updates
+
+With the 8.12 release of Elasticsearch, a new link:https://www.elastic.co/guide/en/elasticsearch/reference/8.12/query-dsl-knn-query.html[`knn` query]
+was introduced. This query is more aligned with the vector search capabilities provided by Hibernate Search,
+and integration with Elasticsearch was updated to use this new query.
+This means that previous limitations imposed on the vector search when using the Elastic distribution of the Elasticsearch backend
+are now removed. It also means that the vector search on this distribution will only be available starting with the 8.12.0 version.
```
Comment (yrodiere) [inline posts/MarkoB/2024-02-12-hibernate-search-7-1-0-CR1.adoc]:
```suggestion
are now removed.

It also means that vector search on Elasticsearch is now only available with Elasticsearch 8.12 and later; Hibernate Search no longer supports vector search on Elasticsearch 8.11 and earlier.
```

File: posts/MarkoB/2024-02-12-hibernate-search-7-1-0-CR1.adoc [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+See this section of the reference documentation on the link:{hsearch-doc-url-prefix}#search-dsl-predicate-query-string[`queryString` predicate]
+for more information.
+
+[[other-changes]]
+=== Other improvements and bug fixes
+
+* link:{hsearch-jira-url-prefix}/HSEARCH-5020[HSEARCH-5020]:
+Hibernate Search increases the maximum allowed vector dimension for a Lucene backend to `4096` to allow indexing longer vectors produced by some models.
+* link:{hsearch-jira-url-prefix}/HSEARCH-5073[HSEARCH-5073]:
+Hibernate Search opens up an ability to specify link:{hsearch-doc-url-prefix}#search-dsl-predicate-simple-query-string-minimumshouldmatch[minimum-should-match parameter] on a simple query predicate.
```
Comment (yrodiere) [inline posts/MarkoB/2024-02-12-hibernate-search-7-1-0-CR1.adoc]:
```suggestion
Hibernate Search now allows specifying a link:{hsearch-doc-url-prefix}#search-dsl-predicate-simple-query-string-minimumshouldmatch[minimum-should-match parameter] on a simple query predicate.
```

**Review by yrodiere — APPROVED** [review]
Thanks, my comments below. Feel free to merge when you're satisfied.

File: posts/MarkoB/2023-11-30-hibernate-search-7-1-0-Alpha1.adoc (hunk 1 above)
Comment (marko-bekhta) [reply posts/MarkoB/2023-11-30-hibernate-search-7-1-0-Alpha1.adoc]: > Typo: stanadlone => stanadlone.

I've tried ... but nothing got replaced 😅 it's a highly-typo-prone word 😄 (replaced with `standalone` thanks!)

**Comment by marko-bekhta** [comment]
Thanks for taking a look! 😃

**Pull request merged**